lib/handlers/check_version_handler.dart

```diff
@@ -4,80 +4,132 @@ import 'package:breez/bloc/blocs_provider.dart';
+    } else if (!upgrading && error.contains('bad version')) {
+      _handleBadVersion(context);
+    } else {
+      _log.info("Unhandled error $error");
     }
-    if (err.contains('connection error')) {
-      showNoConnectionDialog(context).then((retry) {
-        if (kDebugMode) {
-          print("-- showNoConnectionDialog --\n$retry");
-        }
-        if (retry == true) {
-          Future.delayed(const Duration(seconds: 1), () {
-            checkVersionDialog(context, userProfileBloc);
-          });
-        }
+  });
+}
+
+void _handleUpgrading(
+  BuildContext context,
+  UserProfileBloc userProfileBloc,
+) {
+  _log.info("_handleUpgrading");
+  final texts = context.texts();
+  final themeData = Theme.of(context);
+
+  AccountBloc accBloc = AppBlocsProvider.of<AccountBloc>(context);
+  accBloc.setNodeUpgrading(true);
+
+  showFlushbar(
+    context,
+    isDismissible: false,
+    position: FlushbarPosition.TOP,
+    duration: Duration.zero,
+    messageWidget: Text(
+      texts.handler_check_version_error_upgrading_servers,
+      style: theme.snackBarStyle,
+      textAlign: TextAlign.left,
+    ),
+    buttonText: "",
+    icon: SvgPicture.asset(
+      "src/icon/warning.svg",
+      colorFilter: ColorFilter.mode(
+        themeData.colorScheme.error,
+        BlendMode.srcATop,
+      ),
+    ),
+  );
+}
+
+void _handleConnectionError(
+  BuildContext context,
+  UserProfileBloc userProfileBloc,
+) {
+  _log.info("_handleConnectionError");
+  showNoConnectionDialog(context).then((retry) {
+    _log.info("-- showNoConnectionDialog --\n$retry");
+
+    if (retry == true) {
+      Future.delayed(const Duration(seconds: 1), () {
+        checkVersionDialog(context, userProfileBloc);
       });
-    } else if (!upgrading && err.contains('bad version')) {
-      showFlushbar(
-        context,
-        buttonText: texts.handler_check_version_action_update,
-        onDismiss: () {
-          if (defaultTargetPlatform == TargetPlatform.iOS) {
-            launchUrlString('https://testflight.apple.com/join/wPju2Du7');
-          }
-          if (defaultTargetPlatform == TargetPlatform.android) {
-            launchUrlString(
-                'https://play.google.com/apps/testing/com.breez.client');
-          }
-          return false;
-        },
-        position: FlushbarPosition.TOP,
-        duration: Duration.zero,
-        messageWidget: Text(
-          texts.handler_check_version_message,
-          style: theme.snackBarStyle,
-          textAlign: TextAlign.center,
-        ),
-      );
     }
   });
 }
+
+void _handleBadVersion(
+  BuildContext context,
+) {
+  _log.info("_handleBadVersion");
+  final texts = context.texts();
+
+  showFlushbar(
+    context,
+    buttonText: texts.handler_check_version_action_update,
+    onDismiss: () {
+      _log.info("onDismiss $defaultTargetPlatform");
+      if (defaultTargetPlatform == TargetPlatform.iOS) {
+        launchUrlString('https://testflight.apple.com/join/wPju2Du7');
```

Comment:
Why are we not directing ios users to app store?

Reply:
Good question; do you have context on this @erdemyerebasmaz ?

Reply:
@ademar111190 No, but why not use `Source.IS_INSTALLED_FROM_APP_STORE` & `Source.IS_INSTALLED_FROM_TEST_FLIGHT` for redirects on iOS?

Reply:
Good call out; I'll do it.